### ISSUE TITLE
In sctransport.js missing test if callback is defined

### ISSUE THREAD

**Issue by dirkbonne** [SocketCluster/socketcluster-client#96]
Under a stress test I get:

```
TypeError: Cannot read property 'call' of undefined
    at Emitter.SCTransport._abortAllPendingEventsDueToBadConnection (/home/dbonne/dvp/js/webstack/miniclient/node_modules/socketcluster-client/lib/sctransport.js:189:15)
    at Emitter.SCTransport._onClose (/home/dbonne/dvp/js/webstack/miniclient/node_modules/socketcluster-client/lib/sctransport.js:205:10)
    at Timeout._onTimeout (/home/dbonne/dvp/js/webstack/miniclient/node_modules/socketcluster-client/lib/sctransport.js:269:10)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```
In the code in sctransport.js I see:

      var callback = eventObject.callback;
      delete eventObject.callback;
      callback.call(eventObject, badConnectionError, eventObject);

Looks like "if(callback)" has been forgotten.

best regards

**Comment by jondubois** [comment]
@dirkbonne Which version of the client are you using? What are the steps to reproduce?

**Comment by dirkbonne** [comment]
The version is 9.0.0.

I cannot easily reproduce this. I just assumed from the other parts in sctransport.js that the callback could be undefined.

best regards, Dirk

**Comment by jondubois** [comment]
@dirkbonne Thanks. I think was able to reproduce the same issue, it can happen when the socket disconnects soon after a specific event response callback times out on the client.

**Comment by jondubois** [comment]
@dirkbonne It should be fixed now in v9.0.2. Feel free to reopen this issue if not.

**Comment by dirkbonne** [comment]
Okay thanks very much!
